Week2/js/HomeTask2.js

```diff
@@ -0,0 +1,204 @@
+    "./Assets/bulgaria.png",
+    "./Assets/republic.png",
+    "./Assets/germany.png",
+    "./Assets/hungary.png",
+    "./Assets/ireland.png",
+    "./Assets/netherlands.png"
+  ];
+  // this loop will create 10 items
+  for (var i = 0; i < 10; i++) {
+    let mainContainer = document.createElement("div");
+    mainContainer.setAttribute("class", "flex-container");
+    mainContainer.setAttribute("id", i + 1);
+    let logo = document.createElement("div");
+    logo.setAttribute("class", "logo");
+    let logoImage = document.createElement("img");
+    logoImage.setAttribute("src", logoPaths[i]);
+    logoImage.setAttribute("class", "logo-image");
+    logo.appendChild(logoImage);
+    let content = document.createElement("div");
+    content.setAttribute("class", "content");
+    let title = document.createElement("h3");
+    title.innerText = titleNames[i];
+    let description = document.createElement("p");
+    description.innerText =
+      "We have helped over 120 Fortune 1000 companies in the following" +
+      "industries solve their most complex technology challenges. Now see " +
+      "what we can do for you. We have helped over 120 Fortune 1000 companies" +
+      "in the following industries solve their most complex technology" +
+      "challenges. Now see what we can do for you. We have helped over 120" +
+      "Fortune 1000 companies in the following industries solve their most" +
+      "complex technology challenges. Now see what we can do for you";
+    let clickToReadMore = document.createElement("button");
+    clickToReadMore.setAttribute("value", "Continue Reading");
+    clickToReadMore.setAttribute("class", "continue-reading");
+    clickToReadMore.setAttribute("id", i + 1);
+    clickToReadMore.addEventListener("click", showPoPUp);
+    clickToReadMore.innerText = "Continue Reading";
+    content.appendChild(title);
+    content.appendChild(description);
+    content.appendChild(clickToReadMore);
+    mainContainer.appendChild(logo);
+    mainContainer.appendChild(content);
+    divisions[i] = mainContainer;
+  }
+  return divisions;
+};
+var channels = generateItems();
+
+var createCategories = function() {
+  let selectCategory = document.createElement("select");
+  let channelNames = [
+    "Canada",
+    "Mexico",
+    "UnitedStates",
+    "Austria",
+    "Bulgaria",
+    "Czech Republic",
+    "Germany",
+    "Hungary",
+    "Ireland",
+    "Netherlands"
+  ];
+  selectCategory.setAttribute("class", "dropdown");
+  selectCategory.setAttribute("id", "dropdown");
+  selectCategory.addEventListener("change", displaySelectedItem);
+  var category = document.createElement("option");
+  category.setAttribute("value", "ALL");
+  category.innerText = "ALL";
+  selectCategory.appendChild(category);
+  for (var i = 0; i < 10; i++) {
+    var category = document.createElement("option");
+    category.setAttribute("value", i + 1);
+    category.innerText = channelNames[i];
+    selectCategory.appendChild(category);

```

Comment:
check document.createDocumentFragment to add eleemnts to HTML page. This is a better way to add DOM elements. This avoids DOM reflow for every addition of element. 

Reference: https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment